baseband/vlbi_base/base.py

```diff
@@ -70,6 +71,31 @@ def __init__(self, fh_raw, header0, nchan, bps, complex_data, thread_ids,
 
         self.frames_per_second = frames_per_second
         self.offset = 0
+        self.squeeze = squeeze
+
+    @property
+    def sample_shape(self):
+        """Shape of a data sample (possibly squeezed).
+        """
+        if self.squeeze:
+            if not self._squeezed_shape:
+                dims = np.array(self._sample_shape)
```

Comment:
I think going via `np.array` is needlessly expensive. Maybe just (note: didn't check this, so may not work immediately)
```
field_names = getattr(self._sample_shape, '_fields', None)
if field_names is not None:
    squeezed_shape_cls = namedtuple('SampleShape', 
            ','.join(*[field_name for field_name, dim in zip(field_names, self._sample_shape) if dim > 1]))
else:
    squeezed_shape_cls = tuple
self._squeezed_shape = squeezed_shape_cls(*[dim for dim in self._sample_shape if dim > 1])
```

Reply:
Running a few `timeit` tests, using arrays is only slightly more computationally expensive than list comprehension.  I switched to list comprehension, and agree it's more reasonable to call lower-level machinery here.